framework/src/org/checkerframework/framework/util/element/ElementAnnotationUtil.java

```diff
@@ -78,16 +78,27 @@ public static void applyAllElementAnnotations(
      * understand them as type annotations. In particular, this allows the transition from Java 5
      * declaration annotations to Java 8 type annotations.
      *
-     * <p>There are some caveats to this: the interpretation for declaration and type annotations
-     * differs, in particular for arrays and inner types. See the manual for a discussion.
+     * <p>If the annotation is one of the Checker Framework compatibility annotations, for example
```

Comment:
I think it would make sense to highlight this in the `declaration-annotations-moved` section. It might be useful to have this option of using a type annotation to annotate an array component type.